### PR TITLE
Update pyquante bridge for pyquante2 support

### PR DESCRIPTION
**Pull request by mscho527**
_Related Issue: #886_
_Builds upon: #893_

This PR updates the bridge from `cclib` to `pyquante` so that it works with pyquante2. Both the bridge function and test routine have been updated. Input for the bridge function has been modified into `ccData` object as well in the process to partially address #890.

A future PR that updates `cclib.method.volume` will depend on the changes made in this PR.

### CODE REVIEW

File: cclib/bridge/cclib2pyquante.py [hunk 1]
```diff
@@ -9,23 +9,52 @@
...
+    """Create a PyQuante Molecule from ccData object."""
+    _check_pyquante()
+
+    # Check required attributes.
+    required_attrs = {"atomcoords", "atomnos", "charge", "mult"}
+    missing = [x for x in required_attrs if not hasattr(data, x)]
+
+    if missing:
+        missing = " ".join(missing)
+        raise MissingAttributeError(
```
Comment (langner) [inline cclib/bridge/cclib2pyquante.py:36]: Where is MissingAttributeError defined?

File: cclib/bridge/cclib2pyquante.py [hunk 2]
```diff
@@ -9,23 +9,52 @@
...
+            [(data.atomnos[i], tuple(data.atomcoords[-1][i])) for i in range(len(data.atomnos))], 
+            units="Angstroms",
+            charge=data.charge,
+            multiplicity=data.mult,
+        )
+
+    # For pyquante2
+    elif _found_pyquante2:
+        return molecule(
+            [(data.atomnos[i], data.atomcoords[-1][i][0], data.atomcoords[-1][i][1], data.atomcoords[-1][i][2]) for i in range(len(data.atomnos))],
```
Comment (langner) [inline cclib/bridge/cclib2pyquante.py:53]: Build this up more incrementally in a few lines, so that it's more readable.

File: test/test_method.py [hunk 3]
```diff
@@ -19,8 +19,10 @@
 from .method.testnuclear import *
 from .method.testorbitals import *
 from .method.testpopulation import *
-if sys.version_info[0] == 2:
-    from .method.testvolume import *
+# Test for volume is commented off at the moment due to changes in pyquante bridge.
```
Comment (langner) [inline test/test_method.py:22]: Could you rephrase this so that it's imperative and clearly a TODO?

File: test/bridge/testpyquante.py [hunk 4]
```diff
@@ -10,19 +10,64 @@
...
+        super(pyquante2Test, self).setUp()
+        self._found_pyquante2 = find_package("pyquante2")
+        self.data, self.logfile = getdatafile("Gaussian", "basicGaussian16", ["water_ccsd.log"])
+
+    def test_makepyquante(self):
+        # Test pyquante2 bridge
+        from pyquante2 import molecule, rhf, h2o, basisset
+
+        bfs = basisset(h2o)
+        # Copied from water_ccsd.log
```
Comment (langner) [inline test/bridge/testpyquante.py:58]: Is it possible to read in the file and grab the data instead?

File: cclib/bridge/cclib2pyquante.py (hunk 1 above)
Comment (mscho527) [reply cclib/bridge/cclib2pyquante.py:40]: I've just added it. Thanks for finding this.

File: cclib/bridge/cclib2pyquante.py (hunk 2 above)
Comment (mscho527) [reply cclib/bridge/cclib2pyquante.py]: I've split this into two lines. Thanks.

File: test/test_method.py (hunk 3 above)
Comment (mscho527) [reply test/test_method.py]: I've clarified this comment. Thanks

File: test/bridge/testpyquante.py (hunk 4 above)
Comment (mscho527) [reply test/bridge/testpyquante.py:58]: From what I know, `pyquante2` only has xyz reader as a means of directly reading in files. I suppose that it is possible to read in from xyz files in turbomole 7.2 data files, although I am not sure if this is what we want given that parsing turbomole is still under development. (`atomcoords` and `atomnos` are already fully implemented for turbomole parser. `atomcharges` isn't, but it shouldn't be an issue here.)

File: cclib/bridge/cclib2pyquante.py [hunk 5]
```diff
@@ -1,31 +1,66 @@
...
+
+    if missing:
+        missing = " ".join(missing)
+        raise MissingAttributeError(
+            "Could not create pyquante molecule due to missing attribute: " + missing
+        )
+
+    # For older PyQuante
+    if _found_pyquante:
+        return Molecule(
```
Comment (berquist) [inline cclib/bridge/cclib2pyquante.py:46]: Do you think we can avoid duplicating function calls by storing the function name in a variable based on the module found and then calling that?

File: cclib/bridge/cclib2pyquante.py [hunk 6]
```diff
@@ -1,31 +1,66 @@
...
+            "notitle",
+            [(data.atomnos[i], tuple(data.atomcoords[-1][i])) for i in range(len(data.atomnos))], 
+            units="Angstroms",
+            charge=data.charge,
+            multiplicity=data.mult,
+        )
+
+    # For pyquante2
+    elif _found_pyquante2:
+        moldesc = numpy.insert(data.atomcoords[-1], 0, data.atomnos, 1).tolist()
```
[the remark this answers is not in the document]
Comment (mscho527) [reply cclib/bridge/cclib2pyquante.py:56]: There is a very small difference -- In old pyquante, the format is: `[(3,( .0000000000, .0000000000, .0000000000)), (1,( .0000000000, .0000000000,1.629912))]` whereas in pyquante2, it is `[(3,.0000000000, .0000000000, .0000000000), (1, .0000000000, .0000000000,1.629912)]`. I'll add this difference as a comment in the code. Thanks.

File: cclib/bridge/cclib2pyquante.py (hunk 5 above)
Comment (mscho527) [reply cclib/bridge/cclib2pyquante.py:46]: I was separating the code thinking that older PyQuante code will be eventually removed when python 2.x.x support is dropped from `cclib`. If that's not happening too soon, I think your suggestion is a great idea.

File: cclib/bridge/cclib2pyquante.py [hunk 7]
```diff
@@ -1,6 +1,6 @@
 # -*- coding: utf-8 -*-
```
[the remark this answers is not in the document]
Comment (mscho527) [reply cclib/bridge/cclib2pyquante.py:1]: Yes -- I've just run black on this. Thanks.

File: cclib/bridge/cclib2pyquante.py (hunk 5 above)
Comment (berquist) [reply cclib/bridge/cclib2pyquante.py:50]: Eh, I don't think it's worth it after all, it's such a small piece of code, and the arguments are just different enough.

File: cclib/bridge/cclib2pyquante.py [hunk 8]
```diff
@@ -1,31 +1,72 @@
...
 # the terms of the BSD 3-Clause License.
 
 """Bridge for using cclib data in PyQuante (http://pyquante.sourceforge.net)."""
 
+import numpy
 from cclib.parser.utils import find_package
 
+
+class MissingAttributeError(Exception):
+    pass
```
Comment (shivupa) [inline cclib/bridge/cclib2pyquante.py:15]: Should this go elsewhere like `cclib/parser/data.py` or something so we can use this in other methods in the future? @langner @berquist

File: cclib/bridge/cclib2pyquante.py [hunk 9]
```diff
@@ -1,31 +1,72 @@
...
+    _check_pyquante()
+
+    # Check required attributes.
+    required_attrs = {"atomcoords", "atomnos"}
+    missing = [x for x in required_attrs if not hasattr(data, x)]
+
+    if missing:
+        missing = " ".join(missing)
+        raise MissingAttributeError(
+            "Could not create pyquante molecule due to missing attribute: " + missing
```
Comment (shivupa) [inline cclib/bridge/cclib2pyquante.py:43]: `... attribute: {}".format(missing)` is a little neater

File: test/bridge/testpyquante.py (hunk 4 above)
Comment (shivupa) [reply test/bridge/testpyquante.py:58]: I'm going to mark this resolved

File: cclib/bridge/cclib2pyquante.py (hunk 5 above)
Comment (shivupa) [reply cclib/bridge/cclib2pyquante.py:50]: I'm going to mark this resolved

File: cclib/bridge/cclib2pyquante.py (hunk 8 above)
Comment (mscho527) [reply cclib/bridge/cclib2pyquante.py:15]: Just an fyi, I've copied this from `cclib.method.calculationmethod` -- most methods use this as well.

File: cclib/bridge/cclib2pyquante.py (hunk 9 above)
Comment (mscho527) [reply cclib/bridge/cclib2pyquante.py]: I've updated just updated this. Thanks for the suggestion!
